### PR TITLE
User can only delete their own posts

### PR DESCRIPTION
**Pull request by Danielandro**
* added a conditional to view all posts page - user only sees edit/delete buttons for their own posts
* feature test - checks user sees links for their own posts + not for others

### CODE REVIEW

File: app/views/posts/display_all.html.erb [hunk 1]
```diff
@@ -6,6 +6,10 @@
   <br>
   <%= post.user_email %>
   <br>
+  <% if current_user.id == post.user_id %>
+    <p><%= link_to 'Delete', [current_user, post], :method => :delete  %></p>
```
Comment (rachjgriff) [inline app/views/posts/display_all.html.erb]: nice!

File: spec/features/user_can_delete_only_their_posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+require 'pry'
+
+RSpec.feature "Delete & edit only my posts", type: :feature do
```
Comment (rachjgriff) [inline spec/features/user_can_delete_only_their_posts_spec.rb]: WHOAH THIS IS A BEAST!!

File: spec/features/user_can_delete_only_their_posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  scenario "user only sees edit & delete links for their posts" do
+    # User 1 - sign in, create a post, sign out
+    sign_up("user1@gmail.com", "123456")
+    click_link "View Your Posts"
+    create_a_post("I love monsieur Travis")
+    click_link "Sign out"
+    # User 2 - sign in, create a post, view all posts
+    sign_up("user2@gmail.com", "123456")
+    click_link "View Your Posts"
+    create_a_post("J'adore la baguette")
```
Comment (rachjgriff) [inline spec/features/user_can_delete_only_their_posts_spec.rb]: Loving the French! Tres bien!

File: spec/features/user_can_delete_only_their_posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    click_link "View All Posts"
+    # grab posts from database
+    post1 = Post.all.where(message: "I love monsieur Travis")[0]
+    post2 = Post.all.where(message: "J'adore la baguette")[0]    
+
```
Comment (rachjgriff) [inline spec/features/user_can_delete_only_their_posts_spec.rb]: this looks pretty cool - would love an explanation in our app run through catch up

File: spec/features/user_can_delete_only_their_posts_spec.rb (hunk 3 above)
Comment (ajmpawlik) [reply spec/features/user_can_delete_only_their_posts_spec.rb]: Very nice with the comments!

File: spec/features/user_can_delete_only_their_posts_spec.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    post1 = Post.all.where(message: "I love monsieur Travis")[0]
+    post2 = Post.all.where(message: "J'adore la baguette")[0]    
+    
+    expect(page).to have_content("I love monsieur Travis")
+    expect(page).to have_content("J'adore la baguette")
+    expect(page).not_to have_link("Edit", href: "/users/#{post1.user_id}/posts/#{post1.id}/edit")
+    expect(page).not_to have_link("Delete", href: "/users/#{post1.user_id}/posts/#{post1.id}")
+    expect(page).to have_link("Edit", href: "/users/#{post2.user_id}/posts/#{post2.id}/edit")    
+    expect(page).to have_link("Delete", href: "/users/#{post2.user_id}/posts/#{post2.id}")
+  end
```
Comment (LucyMHall) [inline spec/features/user_can_delete_only_their_posts_spec.rb]: Love it :)

File: spec/features/user_can_delete_only_their_posts_spec.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    click_link "View Your Posts"
+    create_a_post("J'adore la baguette")
+    click_link "View All Posts"
+    # grab posts from database
+    post1 = Post.all.where(message: "I love monsieur Travis")[0]
+    post2 = Post.all.where(message: "J'adore la baguette")[0]    
+    
+    expect(page).to have_content("I love monsieur Travis")
+    expect(page).to have_content("J'adore la baguette")
+    expect(page).not_to have_link("Edit", href: "/users/#{post1.user_id}/posts/#{post1.id}/edit")
```
Comment (zepherine2006DEV) [inline spec/features/user_can_delete_only_their_posts_spec.rb]: very clever!

**Pull request merged**